### ISSUE TITLE
Question about MoCa（Multi-mOdality Cut and pAste）(Data Augmentation)

### ISSUE THREAD

**Issue by FANGZHENGFA** [open-mmlab/mmdetection3d#1143]
Where is the relative code of MOCA？
I just find 
        "T" stands for translation;
        "S" stands for scale;
        "R" stands for rotation;
        "HF" stands for horizontal flip;
        "VF" stands for vertical flip.
in  ```transformation_3d_flow```.
Is the ```MOCA``` codes not open source？

**Comment by ZCMax** [comment]
Thanks for your attention, I cannot promise the time of releasing Moca, but it should be merged in a few months due to time problem of the author. Actually, I recommend you to refer to the PR #327 and make some modifications if you have enough time. As far as I know, one of our codebase users has down like this, and achieve the results in moca paper.  Sorry for the inconvenience.

**Comment by FANGZHENGFA** [comment]
> Thanks for your attention, I cannot promise the time of releasing Moca, but it should be merged in a few months due to time problem of the author. Actually, I recommend you to refer to the PR #327 and make some modifications if you have enough time. As far as I know, one of our codebase users has down like this, and achieve the results in moca paper. Sorry for the inconvenience.

Thank you for your reply！
Do you mean that I can modify the code in ```https://github.com/ZwwWayne/mmdetection3d/tree/moca-kitti``` to achieve the results in moca paper?

**Comment by ZCMax** [comment]
Yes, some modifications need be down by yourself to achieve the results in paper.

**Comment by FANGZHENGFA** [comment]
> Yes, some modifications need be down by yourself to achieve the results in paper.

Got it. Thank you very much!

**Comment by FANGZHENGFA** [comment]
Hi.
Could you provide more details about preparing dataset files, such as the steps to generate the ```kitti_mm_dbinfos_train.pkl```file and the file tree structure of KITTI dataset.

**Comment by EchoQiHeng** [comment]
> Hi. Could you provide more details about preparing dataset files, such as the steps to generate the `kitti_mm_dbinfos_train.pkl`file and the file tree structure of KITTI dataset.

I according to this paper found the **instance-train. json** file of KINS dataset, then **Python tools/create_data.py --root-path --out_path --extra-tag kitti --version=mask**. Ultimately, **kitti_infos_xx.pkl** and **kitti_dbinfos_train.pkl** are generated, and **xx.bin.mask.png** and **xx.bin.png** are generated in **kitti_gt_dataset** file. But I don't know where the code is to generate  **kitti_mm_dbinfos_train.pkl**.
If it's convenient for you, Could you please share the **kitti_mm_dbinfos_train.pkl** with me, or tell me the details of generating the **kitti_mm_dbinfos_train.pkl** from the **instance-train.json** file.

**Comment by demerzel1** [comment]
Hi,

Is there any update on releasing Moca? @ZwwWayne

**Comment by Byte247** [comment]
I still wonder if this will ever be released....
